### ISSUE TITLE
Solve MultiSnake

### ISSUE THREAD

**Issue by cswinter** [entity-neural-network/incubator#51]

**Comment by cswinter** [comment]
Trying to run MultiSnake currently crashes because there are several places where we are not correctly handling multiple actors.

**Comment by cswinter** [comment]
https://github.com/entity-neural-network/incubator/pull/91 allows for training with multiple actors. MultiSnake still cannot learn good policies due to lack of https://github.com/entity-neural-network/incubator/issues/9.

**Comment by cswinter** [comment]
https://github.com/entity-neural-network/incubator/issues/9 has landed. Solving MultiSnake is still challenging. There are two parameters along which we can make the environment easier: `max_snake_length` (default 11) and `num_snakes` (default 2).

We can already get close solving `{"num_snakes": 1, "max_snake_length": 2}` with this command (~0.98 at 2dd0716, [W&B](https://wandb.ai/entity-neural-network/enn-ppo/runs/1hvzzjif)):
 
```
run python enn_ppo/enn_ppo/train.py --total-timesteps=1000000 --track --gym-id=MultiSnake --num-envs=64 --num-steps=32 --n-layer=1 '--env-kwargs={"num_snakes": 1, "max_snake_length": 2}'
```

It really struggles with longer snakes though. I verified that you can fully solve even `{"num_snakes": 1, "max_snake_length": 11}` with a hacky version of https://github.com/entity-neural-network/incubator/issues/104. To solve `{"num_snakes": 2, "max_snake_length": 11}`, we will likely need better positional encodings and transformer (https://github.com/entity-neural-network/incubator/issues/10).

**Comment by cswinter** [comment]
Critical bug fixed in https://github.com/entity-neural-network/incubator/pull/121

**Comment by cswinter** [comment]
Solution for `{"num_snakes": 1, "max_snake_length": 11}`: https://wandb.ai/entity-neural-network/enn-ppo/reports/VmlldzoxNDI1NDQ5

**Comment by cswinter** [comment]
New record for `"num_snakes": 2` using relative positional encoding with per-entity values and gradient accumulation for large batch sizes: https://wandb.ai/entity-neural-network/enn-ppo/runs/19fk5i7w

Still only reaches ~0.95 average episodic return after 250M training steps (11 hours of training). I think this task might simply be hard to solve since it requires planning ahead by multiple timesteps, but there could also still be a limitation in our positional encoding.

**Comment by cswinter** [comment]
Another [bug](https://github.com/entity-neural-network/incubator/pull/215) found, score of 0.985 is now possible with 100M training steps: https://wandb.ai/entity-neural-network/enn-ppo/reports/MultiSnake-2-snakes-11-length--VmlldzoxNzgwNzEw
Score of 0.99 should be doable now.
